src/xpath.c

```diff
@@ -5514,8 +5516,14 @@ moveto_node_check(struct lyd_node *node, enum lyxp_node_type root_type, const ch
     }
 
     /* name check */
-    if (strcmp(node_name, "*") && !ly_strequal(node->schema->name, node_name, 1)) {
-        return -1;
+    if (node_name_dict) {
+        if (strcmp(node_name_dict, "*") && !ly_strequal(node->schema->name, node_name_dict, 1)) {
+            return -1;
+        }
+    } else {
+        if (strcmp(node_name, "*") && strncmp(node->schema->name, node_name, node_name_len)) {
```

Comment:
Be careful here, `strncmp` is not enough, the exact length (ending 0) must also be checked.